src/schemas/Entity.js

```diff
@@ -61,23 +61,17 @@ export default class EntitySchema {
     return this.getId(input, parent, key);
   }
 
-  denormalize(entityOrId, unvisit, getDenormalizedEntity) {
-    const entity = getDenormalizedEntity(this, entityOrId);
-    if (typeof entity !== 'object' || entity === null) {
-      return entity;
-    }
-
+  denormalize(entity, unvisit) {
     if (ImmutableUtils.isImmutable(entity)) {
-      return ImmutableUtils.denormalizeImmutable(this.schema, entity, unvisit, getDenormalizedEntity);
+      return ImmutableUtils.denormalizeImmutable(this.schema, entity, unvisit);
     }
 
-    const processedEntity = { ...entity };
```

Comment:
This was where we were preventing mutations of the entities in the store which was creating a new reference each time an entity was denormalized. This is now just done once when an entity is first "unvisited", so any subsequent hits will have the same exact reference.